tests/unity/CMakeLists.txt

```
@@ -198,16 +204,25 @@ endfunction()
 #for example if file under test is include mocked header as <foo/header.h> then
 # mock and replaced header should be placed in <mock_path>/foo with <mock_path>
 # added as include path.
-# EXCLUDE <pattern>: Exclude functions matching pattern. Pattern can be a simple style regex.
+# EXCLUDE <pattern>: Exclude functions matching pattern. Deprecated, use FUNC_EXCLUDE instead.
+# FUNC_EXCLUDE <pattern>: Exclude functions matching pattern. Pattern can be a simple style regex.
+# WORD_EXCLUDE <pattern>: Exclude words matching pattern. Pattern can be a simple style regex.
 function(cmock_handle header_file)
-  cmake_parse_arguments(CMOCK "" "" "EXCLUDE" ${ARGN})
+  cmake_parse_arguments(CMOCK "" "" "EXCLUDE;FUNC_EXCLUDE;WORD_EXCLUDE" ${ARGN})
   get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
   set(CMOCK_PRODUCTS_DIR ${APPLICATION_BINARY_DIR}/mocks)
 
+  set(exclude_list ${CMOCK_EXCLUDE})
+  list(LENGTH exclude_list num_exclude)
+  if (NOT ${num_exclude} EQUAL 0)
+    message(DEPRECATION " cmock_handle(EXCLUDE) is deprecated, use FUNC_EXCLUDE instead")
+    set(CMOCK_FUNC_EXCLUDE ${CMOCK_EXCLUDE})
+  endif()
```

a simpler approach.

The `CMOCK_EXCLUDE` is only defined if user used it as argument, and two avoid unexpected overwrite in case a caller uses both `CMOCK_EXCLUDE` and `CMOCK_FUNC_EXCLUDE` as arguments, then combine the lists.

```suggestion
  if (DEFINED CMOCK_EXCLUDE)
    message(DEPRECATION " cmock_handle(EXCLUDE) is deprecated, use FUNC_EXCLUDE instead")
    list(APPEND CMOCK_FUNC_EXCLUDE ${CMOCK_EXCLUDE})
  endif()
```